### PR TITLE
Added functionality to abort request on server and client side

### PR DESCRIPTION
**Pull request by mczapkowicz**
[FEE-434](https://metakeen.atlassian.net/jira/software/projects/FEE/boards/10?selectedIssue=FEE-434)

### CODE REVIEW

File: lib/record-events-browser.js [hunk 1]
```diff
@@ -6,316 +6,319 @@ import each from 'keen-core/lib/utils/each';
...
-      // Process "events" transform pipeline
-      getExtendedEventBody(extendedEventBody, self.extensions.events);
-      // Process "collection" transform pipeline
-      getExtendedEventBody(extendedEventBody, self.extensions.collections[eventCollection]);
-      // Blend existing eventBody data into the result
-      getExtendedEventBody(extendedEventBody, [eventBody]);
-      // Push extendedEventBody into new hash
-      extendedEventsHash[eventCollection].push(extendedEventBody);
+export function recordEvents(eventsHash, callback) {
+    console.log('record')
```
Comment (apryka) [inline lib/record-events-browser.js:100]: please remove unnecessary log

**Review by apryka — COMMENTED** [review]
existing tests are failing

File: lib/record-events-browser.js (hunk 1 above)
Comment (maciejrybaniec) [reply lib/record-events-browser.js:100]: +1

Comment (dariuszlacheta) [reply lib/record-events-browser.js:100]: +1

**Comment by stof** [comment]
This implementation is broken. The promise on which an `abort()` method is added is not the promise returned by the client public API, due to the additional chaining being done to parse the json body of the response.

Such API using a custom promise implementation is very easy to break when using chaining or await (this issue in the public API is the proof btw). It might be better to add a `signal` option in the argument instead, allowing to abort the request with the AbortController/AbortSignal API used in modern DOM APIs, as it is a lot less fragile (the signal can be passed around in calls without loosing it in async calls)

**Pull request merged**